### DISCUSSION TITLE
Using Sonos with sound bar and TV, add TV as a source

### DISCUSSION THREAD

**Discussion by harryfine** [music-assistant/support#2290] [Feature requests and ideas]
I'm assuming that almost all Sonos users have a sound bar (Arc, Beam, Playbar) with a TV playing through it using either the HDMI or optical port.  Using the old Sonos app, when the TV turns on, the source changes to TV automatically on the sound bar attached to the TV.  This assumes you have the option set in Sonos called TV autoplay and optionally, ungroup on autoplay.  It's slick.

But sometimes the sound bar doesn’t auto-switch to TV when it senses the HDMI or optical sound source. With the Sonos app you could select the sound bar and then select the media source as TV.

Last night I was listening to music through MA, and then decided to watch the Blue Jay game.  In MA, it switched to showing external source on the sound bar, the sound bar (called Living Room Speaker Pair) came to life with audio from the TV and I could adjust the volume in MA.  Perfect.  No need for the terrible Sonos app except to setup new products.

But playing with the volume or pairing controls, I accidentally disconnected the external source from the Living Room speaker.  But now, the Living Room speaker pair speaker was gone.  And I couldn't get it back without going into the HA Player Provider settings and re-synchonizing.  After that, the speaker re-appeared.

So what I’m saying is that without a way to manually select TV as a sound source for a sound bar, I would still need the Sonos app to do that, and I think to make the Sonos experience more complete, there should be either a feature or bug fix to address this.

**Comment by rymiller11** [comment]
I get what you're saying.  A TV should be a Music Provider that can be played back on any combination of Player Providers in MA.  For that matter, sources like Roku, AppleTV, Amazon Firestick, etc. could all be Music Providers (or, perhaps a new category called "TV Providers" that could be paired with whatever Player Provider.

My use case is this:  I have a Dodgers game on my TV via the Roku and can see the game on my TV across the room, outside my patio doors all the way to my hottub.  However, it's only playing on my ceiling Sonos speakers.  I would like to select my Hottub Sonos speakers as a player provider in MA.  Make sense?  I can then hear the game all the way in my backyard.

**Comment by harryfine** [comment]
I have a similar issue.  Last week I had the TV on very early in the
morning watching the British Open golf tournament.  The TV plays
through the Sonos Arc and some rear speakers.

My wife came downstairs later, and the morning music routine ran based
on the time/motion/day/binary sensor.  So our morning radio station
started playing out of the Sonos Arc speaker instead of golf from the
TV.

So, I had 2 choices.  

1.  Turn the TV off and on, which usually works, or: 
2.  Use the Sonos app which I never use for anything else, to re-select
TV as the source for my Sonos Arc. 

It would have been nice to just re-select TV as the source for my Sonos
Arc in Music Assistant.

-----Original Message-----
From: rymiller11 ***@***.***>
Reply-To: music-assistant/support
***@***.***>
To: music-assistant/support ***@***.***>
Cc: Harry Fine ***@***.***>, Author ***@***.***>
Subject: Re: [music-assistant/support] Using Sonos with sound bar and
TV, add TV as a source (Discussion #2290)
Date: 2025-07-22 06:34:30 PM

I get what you're saying. A TV should be a Music Provider that can be
played back on any combination of Player Providers in MA. For that
matter, sources like Roku, AppleTV, Amazon Firestick, etc. could all be
Music Providers (or, perhaps a new category called "TV Providers" that
could be paired with whatever Player Provider.
My use case is this: I have a Dodgers game on my TV via the Roku and
can see the game on my TV across the room, outside my patio doors all
the way to my hottub. However, it's only playing on my ceiling Sonos
speakers. I would like to select my Hottub Sonos speakers as a player
provider in MA. Make sense? I can then hear the game all the way in my
backyard.
—
Reply to this email directly, view it on GitHub [1], or unsubscribe
[2].
You are receiving this because you authored the thread.Message ID:
<music-assistant/support/repo-
***@***.***>


[1] view it on GitHub
    https://github.com/orgs/music-assistant/discussions/2290#discussioncomment-13855839
[2] unsubscribe
